src/main/java/org/assertj/core/api/AbstractOffsetDateTimeAssert.java

```diff
@@ -671,6 +671,33 @@ public SELF isStrictlyBetween(String startExclusive, String endExclusive) {
     return isStrictlyBetween(parse(startExclusive), parse(endExclusive));
   }
 
+  /**
+   * Verifies that actual and given {@code OffsetDateTime} have the same {@link java.time.Instant}.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // successful assertions
+   * OffsetDateTime OffsetDateTime1 = OffsetDateTime.of(2000, 12, 12, 3, 0, 0, 0, ZoneOffset.ofHours(3));
+   * OffsetDateTime OffsetDateTime2 = OffsetDateTime.of(2000, 12, 12, 0, 0, 0, 0, ZoneOffset.ofHours(0));
+   * assertThat(OffsetDateTime1).hasSameInstantAs(OffsetDateTime2);
+   *
+   * // failing assertions
+   * OffsetDateTime OffsetDateTime1 = OffsetDateTime.of(2000, 12, 12, 3, 0, 0, 0, ZoneOffset.ofHours(3));
+   * OffsetDateTime OffsetDateTime2 = OffsetDateTime.of(2000, 12, 12, 2, 0, 0, 0, ZoneOffset.ofHours(0));
+   * assertThat(OffsetDateTime1).hasSameInstantAs(OffsetDateTime2);</code></pre>
+   *
+   * @param other the given {@link java.time.OffsetDateTime}.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code OffsetDateTime} is {@code null}.
+   * @throws IllegalArgumentException if other {@code OffsetDateTime} is {@code null}.
+   * @throws AssertionError if the actual {@code OffsetDateTime} does not have the same {@code Instant} as the other.
+   */
+  public SELF hasSameInstantAs(OffsetDateTime other) {
+    Objects.instance().assertNotNull(info, actual);
+    assertOffsetDateTimeParameterIsNotNull(other);
+    comparables.assertEqual(info, actual.toInstant(), other.toInstant());
```

Comment:
It would be better to introduce a specific error message stating we were comparing the OffsetDateTime as Instant.
This means adding a `ShouldBeAtSameInstant` with an error message like:
```
"%nExpecting%n" +
"  <%s>%n" +
"to be at the same instant as:%n" +
"  <%s>%n" +
"but actual instance was%n" +
"  <%s>%n" +
"and expected instant was:%n" +
"  <%s>"
```
A `ShouldBeAtSameInstant_create_Test` should be added.

You cab refer to `ShouldHaveSameHashCode` and `ShouldHaveSameHashCode_create_Test` for how to do this

Reply:
This also means `comparables.assertEqual` can't be used anymore, but something like:
```
if (!comparables.toInstant().equals(other.toInstant())) throw failures.failure(info, shouldBeAtSameInstant(actual, other));
```

Reply:
Added a `ShouldBeAtSameInstant`, using it as the failure message for `isAtSameInstantAs`.